### PR TITLE
DM-151 User profile page's Organizations tab

### PR DESCRIPTION
**Pull request by dhruvio**
Includes back-end changes and bugfixes (@sutherlanda please review back-end changes).

### CODE REVIEW

File: src/back-end/lib/db.ts [hunk 1]
```diff
@@ -351,35 +352,42 @@ export async function readOneAffiliation(connection: Connection, user: Id, organ
 
 export async function readOneAffiliationById(connection: Connection, id: Id): Promise<Affiliation> {
   const result = await connection('affiliations')
-    .where({
-      id
-    })
+    .where({ id })
+    .andWhereNot({ membershipStatus: MembershipStatus.Inactive })
```
Comment (dhruvio) [inline src/back-end/lib/db.ts]: @sutherlanda FYI this `andWhereNot` function call was missing in several affiliation functions in `db.ts`

Comment (dhruvio) [reply src/back-end/lib/db.ts]: It resulted in a bug that would vend inactive affiliations to API consumers.

File: src/back-end/lib/resources/affiliation.ts [hunk 2]
```diff
@@ -182,7 +182,7 @@ const resource: Resource = {
         }
 
         const existingAffiliation = validatedAffiliationId.value;
-        if (existingAffiliation.membershipType === MembershipType.Owner && await readActiveOwnerCount(connection, request.query.organization) === 1 ) {
+        if (existingAffiliation.membershipType === MembershipType.Owner && await readActiveOwnerCount(connection, existingAffiliation.organization.id) === 1 ) {
```
Comment (dhruvio) [inline src/back-end/lib/resources/affiliation.ts]: @sutherlanda I think you forgot to change this ref from the query string to the `existingAffiliation`. Let me know if the query param ref was intentional.

File: src/shared/lib/resources/affiliation.ts [hunk 3]
```diff
@@ -24,7 +24,8 @@ export interface Affiliation {
 
 // Used when returning a list of the current user's affiliations
 export interface AffiliationSlim {
-  organizationName: string;
+  id: Id;
+  organization: Pick<Organization, 'id' | 'legalName'>;
```
Comment (dhruvio) [inline src/shared/lib/resources/affiliation.ts]: @sutherlanda FYI I expanded the `AffiliationSlim` type with necessary fields for the front-end.

File: src/back-end/lib/db.ts (hunk 1 above)
Comment (sutherlanda) [reply src/back-end/lib/db.ts]: Got it, thanks for correcting.

File: src/back-end/lib/resources/affiliation.ts (hunk 2 above)
Comment (sutherlanda) [reply src/back-end/lib/resources/affiliation.ts]: Yes that was an oversight, thank you.

**Pull request merged**